demo/summer-2016.html

```diff
@@ -244,11 +244,41 @@
         langTools.addCompleter(texCompleter);
         console.log("Simple Latex Completer successfully added");
     });
-    
-    document.getElementById("editor").addEventListener("click", function() {
-        highlighter.highlightBrackets(editor);
+
+    var keyboardHandler = {
+      name: 'highlightBrackets',
+      bindKey: {win: 'Ctrl-Shift-9', mac: 'Command-Shift-9'},
+      exec: function(editor) {return highlighter.highlightBrackets(editor)},
+      readOnly: true
+    }
+    editor.commands.addCommand(keyboardHandler);
+
+
+    function removeHighlightBracket() {     
+        session.removeMarker(session.$bracketHighlightLeft);
+        session.removeMarker(session.$bracketHighlightRight);
+        session.$bracketHighlightLeft = null;
+        session.$bracketHighlightRight = null;
+    }
+
+    session.getSelection().on("changeCursor", function() {
+      if (session.$bracketHighlightLeft || session.$bracketHighlightRight) {
+        removeHighlightBracket();
+        if(!session.$bracketHighlight)
+            if (!isOutFromAreaOfHighlight() && !session.$bracketHighlight)
+            highlighter.highlightBrackets(editor);
+      }   
     });
 
+    function isOutFromAreaOfHighlight() {
+      var oldPosition = {};
+      for (var key in session.$positionOfHighlight)
```

Comment:
If you run 'findSurroundingBrackets' that changes $positionOfHighlight of its value, therefore I save  oldPosition by value.